src/include/serial_inline.h

```diff
@@ -296,11 +299,16 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
     if (WT_PAGE_TRYLOCK(session, page) != 0)
         return (0);
 
-    obsolete = __wt_update_obsolete_check(session, page, upd->next, true);
+    obsolete = __wt_update_obsolete_check(session, page, upd->next, true, &upd_count);
 
     WT_PAGE_UNLOCK(session, page);
 
     __wt_free_update_list(session, &obsolete);
 
+    if (upd_count > 1000) {
+        WT_STAT_CONN_INCR(session, cache_eviction_force_update_list);
+        __wt_page_evict_soon(session, cbt->ref);
```

Comment:
Are we guaranteed that cbt->ref will not be NULL. I would think so, this being a serial_update function, but @agorrod can you confirm ?